app/src/main/java/org/stepic/droid/ui/dialogs/LoadingProgressDialog.kt

```diff
@@ -0,0 +1,26 @@
+package org.stepic.droid.ui.dialogs
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.support.annotation.StringRes
+import android.view.Window
+import kotlinx.android.synthetic.main.dialog_progress.*
+
+import org.stepic.droid.R
+
+
+class LoadingProgressDialog
+@JvmOverloads constructor(
+        context: Context,
+        @StringRes private val titleRes: Int = R.string.loading) : Dialog(context) {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        requestWindowFeature(Window.FEATURE_NO_TITLE)
+        setContentView(R.layout.dialog_progress)
+        progressDialogTitle.setText(titleRes)
+        progressDialogMessage.setText(R.string.loading_message)
+        setCancelable(false)
```

Comment:
Let's add cancelable on touch outside if we want to do this dialog not cancelable. Partially cancelable behaviour is strange. Isn't it?